### PR TITLE
simplifying updateDatabase function to prevent timeout bug

### PR DESCRIPTION
**Pull request by aholachek**
hi @romanchyla can you take a look at this and tell me if it seems like an appropriate fix for the problem?

### CODE REVIEW

File: src/js/modules/orcid/orcid_api.js [hunk 1]
```diff
@@ -921,23 +924,24 @@ define([
...
-            }
-          }, 200);
-          setTimeout(function() {
-            clearInterval(tme);
-          }, 3000);
-          return whenDone.promise();
+
+        //update is currently in progress
+        if (this.updateDatabasePromise){
+          return this.updateDatabasePromise
```
Comment (romanchyla) [inline src/js/modules/orcid/orcid_api.js]: you should return this.updateDatabasePromise.promise()

File: src/js/modules/orcid/orcid_api.js [hunk 2]
```diff
@@ -921,23 +924,24 @@ define([
...
+        }
+
+        //otherwise, have to run the update and set this.updateDatabasePromise
+        this.updateDatabasePromise = whenDone;
+
+        //this function will clear the resolved whendone promise after 30 seconds,
+        //so if new requests are made, they will have to update database again
+        function clearWhenDoneAfterDelay(){
+          setTimeout(function(){
+            self.updateDatebasePromise = null;
```
Comment (romanchyla) [inline src/js/modules/orcid/orcid_api.js]: call .reject() first, then set to null

Comment (romanchyla) [inline src/js/modules/orcid/orcid_api.js]: oh wait, why this function? if I remember correctly, if we set the timeout parameter, then this is not necessary

Comment (aholachek) [inline src/js/modules/orcid/orcid_api.js]: what will happen to the requests for _getRecordInfo from other item views that come back after the promise has been resolved? if we remove the deferred right after it returns, they will show errors, right?

**Pull request merged**